### ISSUE TITLE
4D FITS and MultiDim

### ISSUE THREAD

**Issue by nabobalis** [ejeschke/ginga#202]
Creating a 4D FITS file, with shape [10,89,1000,1000] which are [Time,Wavelength,Spatial]. NAXIS3 and NAXIS4 are the correct length in MultiDim but it can not index NAXIS3 beyond the range of NAXIS4.

This seems to me that NAXIS3 and NAXIS4 are switched when they should not be. So somewhere the ordering of the axes is getting incorrectly set but I can not seem to track it down. 

@Cadair @rupak0577

**Comment by Cadair** [comment]
A simple example of this would be:

```python

import numpy as np
from astropy.io import fits

arr = np.random.random([10, 89, 1000, 1000])

hdu = fits.PrimaryHDU(arr)
hdulist = fits.HDUList([hdu])
hdulist.writeto('new.fits')
```

@nabobalis can you check the above?

**Comment by nabobalis** [comment]
@Cadair I already did with some IBIS data. But I can run that.

**Comment by Cadair** [comment]
@nabobalis just to give a way for other people to check it.

**Comment by nabobalis** [comment]
@Cadair I ran it anyway, the same issue.

**Comment by Cadair** [comment]
yes probably best to drop the image down to 100x100

**Comment by Cadair** [comment]
@nabobalis So it seems that:

The callback gets added with the numpy dimension number here: https://github.com/ejeschke/ginga/blob/master/ginga/misc/plugins/MultiDim.py#L255

which then calls: https://github.com/ejeschke/ginga/blob/master/ginga/misc/plugins/MultiDim.py#L210

which calls: https://github.com/ejeschke/ginga/blob/master/ginga/misc/plugins/MultiDim.py#L412

which inverts the axis here: https://github.com/ejeschke/ginga/blob/master/ginga/misc/plugins/MultiDim.py#L423

so I guess there is a bug in that axis inversion code at the end there.

**Comment by rupak0577** [comment]
Hmm, I should have reported this before - I had this issue with a 4D file but thought it was due to the file. Also, another issue is the slider control buttons which seem to work only for NAXIS3.

**Comment by ejeschke** [comment]
Good find!  I'll take a closer look later today, but if you have a fix, please send a PR!

**Comment by rupak0577** [comment]
I have been trying to fix it by adding checkboxes next to the axes but had some difficulty in following the code. Nevertheless, will try again. Also, I am confused by [this](https://github.com/ejeschke/ginga/blob/master/ginga/misc/plugins/MultiDim.py#L414) - it seems the option to choose the axis is already present. Or is it for something else?

**Comment by ejeschke** [comment]
Fixed via #293
